pkg/syncer/syncertest/fake/subresource_storage.go

```diff
@@ -44,11 +44,11 @@ func (ss *SubresourceStorage) setSubresourceInterface(uObj *unstructured.Unstruc
 }
 
 // Update the sub-resource field. All other fields are ignored.
-func (ss *SubresourceStorage) Update(ctx context.Context, obj client.Object, opts *client.UpdateOptions) error {
+func (ss *SubresourceStorage) Update(ctx context.Context, obj client.Object, opts *client.SubResourceUpdateOptions) error {
 	ss.Storage.lock.Lock()
 	defer ss.Storage.lock.Unlock()
 
-	err := ss.Storage.validateUpdateOptions(opts)
+	err := ss.Storage.validateUpdateOptions(&opts.UpdateOptions)
```

Comment:
Might be slightly cleaner to have a Storage.validateSubresourceUpdateOptions that takes the whole opts and handles delegating the validation of UpdateOptions to validateUpdateOptions

Reply:
done